### ISSUE TITLE
./configure 0.8.8 (and 0.9) openssl errors

### ISSUE THREAD

**Issue by piotrkochan** [redis/RedisDesktopManager#3934]
RDM Version: 0.8.8
Environment (OS name and version):  

```
Distributor ID:	Ubuntu
Description:	Ubuntu 16.04.3 LTS
Release:	16.04
Codename:	xenial
```

Redis Server Version: -

./configure error:

```
[ 24%] Linking C executable example-sftp_write_nonblock
../src/libssh2.a(global.c.o): In function `libssh2_init':
global.c:(.text+0x20): undefined reference to `OPENSSL_add_all_algorithms_noconf'
global.c:(.text+0x25): undefined reference to `OpenSSL_add_all_ciphers'
../src/libssh2.a(openssl.c.o): In function `_libssh2_cipher_init':
openssl.c:(.text+0x4dd): undefined reference to `EVP_CIPHER_CTX_init'
../src/libssh2.a(openssl.c.o): In function `_libssh2_sha1_init':
openssl.c:(.text+0xa87): undefined reference to `EVP_MD_CTX_init'
../src/libssh2.a(openssl.c.o): In function `_libssh2_sha1':
openssl.c:(.text+0xad4): undefined reference to `EVP_MD_CTX_init'
../src/libssh2.a(openssl.c.o): In function `_libssh2_sha256_init':
openssl.c:(.text+0xb5d): undefined reference to `EVP_MD_CTX_init'
../src/libssh2.a(openssl.c.o): In function `_libssh2_sha256':
openssl.c:(.text+0xbaa): undefined reference to `EVP_MD_CTX_init'
../src/libssh2.a(openssl.c.o): In function `_libssh2_md5_init':
openssl.c:(.text+0xc33): undefined reference to `EVP_MD_CTX_init'
../src/libssh2.a(openssl.c.o): In function `_libssh2_pub_priv_keyfilememory':
openssl.c:(.text+0x144b): undefined reference to `OpenSSL_add_all_ciphers'
../src/libssh2.a(mac.c.o): In function `mac_method_hmac_sha2_512_hash':
mac.c:(.text+0xe6): undefined reference to `HMAC_CTX_init'
mac.c:(.text+0x1a8): undefined reference to `HMAC_CTX_cleanup'
../src/libssh2.a(mac.c.o): In function `mac_method_hmac_sha2_256_hash':
mac.c:(.text+0x234): undefined reference to `HMAC_CTX_init'
mac.c:(.text+0x2f6): undefined reference to `HMAC_CTX_cleanup'
../src/libssh2.a(mac.c.o): In function `mac_method_hmac_sha1_hash':
mac.c:(.text+0x382): undefined reference to `HMAC_CTX_init'
mac.c:(.text+0x444): undefined reference to `HMAC_CTX_cleanup'
../src/libssh2.a(mac.c.o): In function `mac_method_hmac_md5_hash':
mac.c:(.text+0x563): undefined reference to `HMAC_CTX_init'
mac.c:(.text+0x625): undefined reference to `HMAC_CTX_cleanup'
../src/libssh2.a(mac.c.o): In function `mac_method_hmac_ripemd160_hash':
mac.c:(.text+0x744): undefined reference to `HMAC_CTX_init'
mac.c:(.text+0x806): undefined reference to `HMAC_CTX_cleanup'
../src/libssh2.a(crypt.c.o): In function `crypt_dtor':
crypt.c:(.text+0x159): undefined reference to `EVP_CIPHER_CTX_cleanup'
collect2: error: ld returned 1 exit status
example/CMakeFiles/example-sftp_write_nonblock.dir/build.make:98: recipe for target 'example/example-sftp_write_nonblock' failed
make[2]: *** [example/example-sftp_write_nonblock] Error 1
CMakeFiles/Makefile2:182: recipe for target 'example/CMakeFiles/example-sftp_write_nonblock.dir/all' failed
make[1]: *** [example/CMakeFiles/example-sftp_write_nonblock.dir/all] Error 2
Makefile:160: recipe for target 'all' failed
make: *** [all] Error 2
```

**Comment by uglide** [comment]
0.8.8 branch is not maintained, please use 0.9

**Comment by piotrkochan** [comment]
Sorry but 0.9 gives similar error.

```
 -lQt5Gui -lQt5Qml -lQt5Network -lQt5Concurrent -lQt5Core -lGL -lpthread 
/usr/local/lib/libssh2.a(knownhost.c.o): In function `knownhost_check':
knownhost.c:(.text+0xa4f): undefined reference to `HMAC_CTX_init'
knownhost.c:(.text+0xaf1): undefined reference to `HMAC_CTX_cleanup'
/usr/local/lib/libssh2.a(openssl.c.o): In function `_libssh2_cipher_init':
openssl.c:(.text+0x4dd): undefined reference to `EVP_CIPHER_CTX_init'
/usr/local/lib/libssh2.a(openssl.c.o): In function `_libssh2_sha1_init':
openssl.c:(.text+0xa87): undefined reference to `EVP_MD_CTX_init'
/usr/local/lib/libssh2.a(openssl.c.o): In function `_libssh2_sha1':
openssl.c:(.text+0xad4): undefined reference to `EVP_MD_CTX_init'
/usr/local/lib/libssh2.a(openssl.c.o): In function `_libssh2_sha256_init':
openssl.c:(.text+0xb5d): undefined reference to `EVP_MD_CTX_init'
/usr/local/lib/libssh2.a(openssl.c.o): In function `_libssh2_sha256':
openssl.c:(.text+0xbaa): undefined reference to `EVP_MD_CTX_init'
/usr/local/lib/libssh2.a(openssl.c.o): In function `_libssh2_md5_init':
openssl.c:(.text+0xc33): undefined reference to `EVP_MD_CTX_init'
/usr/local/lib/libssh2.a(openssl.c.o): In function `_libssh2_pub_priv_keyfilememory':
openssl.c:(.text+0x144b): undefined reference to `OpenSSL_add_all_ciphers'
/usr/local/lib/libssh2.a(global.c.o): In function `libssh2_init':
global.c:(.text+0x20): undefined reference to `OPENSSL_add_all_algorithms_noconf'
global.c:(.text+0x25): undefined reference to `OpenSSL_add_all_ciphers'
/usr/local/lib/libssh2.a(mac.c.o): In function `mac_method_hmac_sha2_512_hash':
mac.c:(.text+0xe6): undefined reference to `HMAC_CTX_init'
mac.c:(.text+0x1a8): undefined reference to `HMAC_CTX_cleanup'
/usr/local/lib/libssh2.a(mac.c.o): In function `mac_method_hmac_sha2_256_hash':
mac.c:(.text+0x234): undefined reference to `HMAC_CTX_init'
mac.c:(.text+0x2f6): undefined reference to `HMAC_CTX_cleanup'
/usr/local/lib/libssh2.a(mac.c.o): In function `mac_method_hmac_sha1_hash':
mac.c:(.text+0x382): undefined reference to `HMAC_CTX_init'
mac.c:(.text+0x444): undefined reference to `HMAC_CTX_cleanup'
/usr/local/lib/libssh2.a(mac.c.o): In function `mac_method_hmac_md5_hash':
mac.c:(.text+0x563): undefined reference to `HMAC_CTX_init'
mac.c:(.text+0x625): undefined reference to `HMAC_CTX_cleanup'
/usr/local/lib/libssh2.a(mac.c.o): In function `mac_method_hmac_ripemd160_hash':
mac.c:(.text+0x744): undefined reference to `HMAC_CTX_init'
mac.c:(.text+0x806): undefined reference to `HMAC_CTX_cleanup'
/usr/local/lib/libssh2.a(crypt.c.o): In function `crypt_dtor':
crypt.c:(.text+0x159): undefined reference to `EVP_CIPHER_CTX_cleanup'
collect2: error: ld returned 1 exit status
Makefile:456: recipe for target '../bin/linux/release/rdm' failed
make: *** [../bin/linux/release/rdm] Error 1
```

**Comment by uglide** [comment]
Openssl is missing on your system.

**Comment by piotrkochan** [comment]
No, it is present.

```
libcurl4-openssl-dev/xenial-updates,xenial-security,now 7.47.0-1ubuntu2.5 amd64 [installed]
libevent-openssl-2.0-5/xenial-updates,xenial-security,now 2.0.21-stable-2ubuntu0.16.04.1 amd64 [installed,automatic]
libgnutls-openssl27/xenial-updates,now 3.4.10-4ubuntu1.4 amd64 [installed]
libio-socket-ssl-perl/xenial,xenial,now 2.024-1 all [installed,automatic]
libnet-smtp-ssl-perl/xenial,xenial,now 1.03-1 all [installed,automatic]
libnet-ssleay-perl/xenial,now 1.72-1build1 amd64 [installed,automatic]
libssl-dev/xenial,now 1.1.0g-2.1+ubuntu16.04.1+deb.sury.org+1 amd64 [installed]
libssl-doc/xenial,xenial,now 1.1.0g-2.1+ubuntu16.04.1+deb.sury.org+1 all [installed,automatic]
libssl1.0.0/xenial-updates,xenial-security,now 1.0.2g-1ubuntu4.10 amd64 [installed]
libssl1.0.2/now 1.0.2m-3 amd64 [installed,local]
libssl1.1/xenial,now 1.1.0g-2.1+ubuntu16.04.1+deb.sury.org+1 amd64 [installed]
openssl/xenial,now 1.1.0g-2.1+ubuntu16.04.1+deb.sury.org+1 amd64 [installed]
python-openssl/xenial,xenial,now 0.15.1-2build1 all [installed,automatic]
ssl-cert/xenial,xenial,now 1.0.37 all [installed,automatic]
```

**Comment by piotrkochan** [comment]
Similar topic https://github.com/voidloop/krakenapi/issues/2#issuecomment-239110348

> The OpenSSL documentation says that function HMAC_CTX_init() was replaced with HMAC_CTX_reset() but in OpenSSL versions 1.1.0.
 
It looks like redis-desktop-manager depends on old openssl package which is not quite elegant.

**Comment by uglide** [comment]
It's libssh2 compilation issue, try to rebuild libssh2

**Comment by artamonovdev** [comment]
Same error

**Comment by shaqman** [comment]
Also getting the same error on 9.1, same environment.
Have any solution been found?

**Comment by codreanulaurentiu** [comment]
Why is this closed? It still doesn't work on 0.9

**Comment by uglide** [comment]
@codreanulaurentiu It's closed because you just need to remove old libssh2.a from `/usr/local/lib/libssh2.a`, and build latest RDM from source to solve this issue

if you still experiencing this issue please provide build log here via gist
